### PR TITLE
Update about page with affiliations

### PR DESCRIPTION
**Pull request by bbengfort**
In response to recent queries about our About page, I've updated it for
clarification -- particularly the part about affiliations. Although this
page will probably change a bit when we ratify the new governance
document, I think this is a start in the right direction. It also takes
care of some of the action items regarding the clarification of our
relationship with DDL.

### CODE REVIEW

File: docs/about.rst [hunk 1]
```diff
@@ -45,28 +45,36 @@ From Wikipedia_:
...
 
+Affiliations
+------------
+
+Yellowbrick is proud to be affiliated with several organizations that provide institutional support to the project. Such support is sometimes financial, often material, and always in the spirit of free and open source software. We can't thank them enough for their role in making Yellowbrick what it is today.
+
+`District Data Labs`_: District Data Labs incubated Yellowbrick and sponsors research labs by purchasing food and organizing events. Research labs are semester long sprints that allow Yellowbrick contributors to meet in person, share a meal, and hack on the project. DDL also sponsors travel to PyCon and PyData conferences for Yellowbrick maintainers and helps us buy promotional material such as stickers and t-shirts.
+
+`NumFOCUS`_: Yellowbrick is a NumFOCUS affiliated project (not a fiscally sponsored project). Our relationship with NumFOCUS has given us a lot of data science cred in the community by being listed on their website. We are also able to apply for small development grants and infrastructure support. We often participate in the project developers mailing list and other activities such as Google Summer of Code.
```
Comment (rebeccabilbro) [inline docs/about.rst]: "We are also able to apply for small development grants and infrastructure support." => "We are also eligible to apply for small development grants and infrastructure support." ?

File: docs/about.rst [hunk 2]
```diff
@@ -45,28 +45,36 @@ From Wikipedia_:
...
+
+`Georgetown University`_: Georgetown primarily provides space for Yellowbrick events including the research labs. Additionally, Georgetown Data Science Certificate students are introduced to Yellowbrick at the beginning of their machine learning education and we often perform user testing of new features on them!
+
+Yellowbrick is developed by volunteers who work on the project in their spare time and not as part of their regular work. If Yellowbrick has become critical to the success of your organization, please consider giving back to Yellowbrick. One way would be to make a financial donation to NumFOCUS on behalf of Yellowbrick. Another would be to contract DDL to give a corporate training on visual machine learning with Yellowbrick. If interested in a more formal affiliate relationship, please get in touch with us directly.
```
Comment (rebeccabilbro) [inline docs/about.rst]: Maybe worthy of an additional subhead? I also like the idea of bulletpointing the suggestions, e.g. 

How to Support Yellowbrick
----------------------------
Yellowbrick is developed by volunteers who work on the project in their spare time and not as part of their regular work. If Yellowbrick has become critical to the success of your organization, please consider giving back to Yellowbrick in one of the following ways:
 - By making a financial donation to NumFOCUS on behalf of Yellowbrick. 
 - By contracting with DDL to give a corporate training on visual machine learning with Yellowbrick. 
 - By getting in touch with us directly if interested in a more formal affiliate relationship!

**Comment by bbengfort** [comment]
@rebeccabilbro thanks for the suggestions - I've implemented them and expanded them further with a new "How to Support Yellowbrick" section that is a bit more extensive; would you mind taking a look at it?

**Comment by rebeccabilbro** [comment]
These edits look great @bbengfort! I particularly like what you've done with the section on suggestions for how to support the project; I feel like I will be pointing people to this page a lot :)

Just synced this branch with `develop`, so will merge as soon as CI tests finish running!

**Pull request merged**